ruby/bindings/InitRubyBindings.cpp

```diff
@@ -874,6 +878,27 @@ void locateEmbeddedGems(bool use_bundler) {
     require 'bundler/plugin'
     #require 'bundler/rubygems_ext'
     require 'bundler/rubygems_integration'
+
+    # Global list, to be populated below
+    $ignore_native_gem_names = []
+
+    module Bundler
+      class RubygemsIntegration
+
+        alias :ori_spec_missing_extensions? :spec_missing_extensions?
+
+        def spec_missing_extensions?(spec, default = true)
+
+          # This avoids getting an annoying message for no reason
+          if $ignore_native_gem_names.any? {|name| name == spec.name }
+            return false
+          end
+
+          return ori_spec_missing_extensions?(spec, default)
+        end
+      end
+    end
```

Comment:
Define a global variable that will store the gem names for which we do have a compatible native extension inside the CLI.

And patch spec_missing_extensions? to just shush the warning that will remain anyways